pkg/virt-controller/watch/application.go

```diff
@@ -634,6 +636,11 @@ func (vca *VirtControllerApp) initCommon() {
 	if err != nil {
 		panic(err)
 	}
+	// Adding a timeout to the clientSet of the migration controller, to avoid potential deadlocks
+	clientSet, err := vca.clientSet.SetRestTimeout(migrationControllerRestTimeout)
```

Comment:
Isn't this client shared by all controllers?

Reply:
Probably yeah, but the timeout is a raw value, not a pointer, so that won't change it for others. It's also the reason why I need to return the modified struct.